service/src/main/java/greencity/service/ShoppingListItemServiceImpl.java

```diff
@@ -481,4 +482,34 @@ private Long deleteUserShoppingListItem(Long id) {
                 .collect(Collectors.toList());
         return getPagebleAdvancedDto(shoppingListItemManagementDtos, shoppingListItems);
     }
+
+    /**
+     * Method adds shopping item by user id and NewShoppingListItemRequestDto.
+     *
+     * @param newShoppingListItemRequestDto {@link NewShoppingListItemRequestDto}
+     *                                      newShoppingListItemRequestDto.
+     * @param userId                        {@link Long} item id.
+     */
+    @Override
+    public void addNewCustomShoppingItem(Long userId, NewShoppingListItemRequestDto newShoppingListItemRequestDto) {
+        customShoppingListItemRepo.save(CustomShoppingListItem.builder()
+            .text(newShoppingListItemRequestDto.getItemDescription())
+            .habit(habitAssignRepo.getOne(newShoppingListItemRequestDto.getHabitAssignId()).getHabit())
```

Comment:
what if `habitAssignRepo.getOne` returns NULL because habit doesn't exist by id? You'll get NPE. please handle this correctly.